bosh-shell/bosh-ssh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+set -eu
+
+if [ $# -eq 0 ]
+then
+    echo "Usage: $(basename "$0") VM/INDEX"
+    exit 1
+fi
+
+target=$1
+
+exec bosh ssh \
+   "$target" \
+   --gateway_host "$BOSH_IP" \
+   --gateway_user vcap \
+   --gateway_identity_file /tmp/bosh_id_rsa
```

Comment:
When running this command, I see lots of deprecation warnings output:
```
/usr/local/bundle/gems/net-ssh-2.9.2/lib/net/ssh/transport/session.rb:67:in `initialize': Object#timeout is deprecated, use Timeout.timeout instead.
/usr/local/bundle/gems/net-ssh-2.9.2/lib/net/ssh/transport/session.rb:84:in `initialize': Object#timeout is deprecated, use Timeout.timeout instead.
/usr/local/bundle/gems/net-ssh-2.9.2/lib/net/ssh/transport/cipher_factory.rb:98: warning: constant OpenSSL::Cipher::Cipher is deprecated
/usr/local/bundle/gems/net-ssh-2.9.2/lib/net/ssh/transport/cipher_factory.rb:72: warning: constant OpenSSL::Cipher::Cipher is deprecated
```

I suspect that this is because the version of bosh-cli that we're using was developed against an earlier version of ruby.  I wonder whether we should use ruby 2.2 for this container, which would match the version used in the bosh-cli container.

Reply:
We made a conscious decision to keep these Ruby warnings as positive pressure to upgrade to modern bosh-cli. Ruby bosh-cli isn't maintained, and [soon Ruby 2.2 won't be either](https://www.ruby-lang.org/en/downloads/).

Reply:
I'm not convinced that's the right choice in this case. Those warnings are a lot of noise, and make it harder to see what's going on when using this.

Secondly, I think it's better that this container is consistent with the one used in the pipeline. At some point we'll have to update all our containers to use a newer ruby, but I think that's better done in one go so that we don't have a mixture of versions in use.

Reply:
Reverted back to 2.2